op-node/rollup/derive/pipeline.go

```diff
@@ -84,15 +85,21 @@ func NewDerivationPipeline(log log.Logger, rollupCfg *rollup.Config, l1Fetcher L
 	// Step stages
 	eng := NewEngineQueue(log, rollupCfg, l2Source, engine, metrics, attributesQueue, l1Fetcher, syncCfg, safeHeadListener)
 
+	// Plasma takes control of the engine finalization signal only when usePlasma is enabled.
+	plasma.OnFinalizedHeadSignal(func(ctx context.Context, ref eth.L1BlockRef) {
```

Comment:
The integration of finalization is kinda weird. As best I can tell, the data path is `recv from state.l1FinalizedSig -> derivation.Finalize -> plasma.Finalize ... (yield for step) then da AdvanceL1Origin -> eng.Finalize`

I think I prefer having the L1 finalization signal arrive to the state. If it needs to be delayed or initiated at at different point, the AdvanceL1Origin function can do that async (via sending a non-blocking event on a subscription).


I think the original flow was `L1 Finalized Signal -> plasma.Da; yield, then da.AdvanceL1Origin -> state.l1FinalizedSig -> derivation.Finalize -> eng.Finalize`. I guess that yield is a little more complex, but it makes the usage of the finalized signal a lot more clear.